config/v1/types_authentication.go

```diff
@@ -19,34 +19,58 @@ type Authentication struct {
 }
 
 type AuthenticationSpec struct {
+	// type identifies the cluster managed, user facing authentication mode in use.
+	// Specifically, it manages the component that responds to login attempts.
+	// The default is IntegratedOAuth.
+	Type AuthenticationType `json:"type"`
+
 	// oauthMetadata contains the discovery endpoint data for OAuth 2.0
 	// Authorization Server Metadata for an external OAuth server.
 	// This discovery document can be viewed from its served location:
 	// oc get --raw '/.well-known/oauth-authorization-server'
 	// For further details, see the IETF Draft:
 	// https://tools.ietf.org/html/draft-ietf-oauth-discovery-04#section-2
-	// If oauthMetadata.name is non-empty, this value has precedence
-	// over the observed value stored in status.oauthMetadata
+	// If oauthMetadata.name is non-empty, this value is used to
+	// calculate the observed value stored in status.oauthMetadata.
+	// Otherwise, observed cluster state may be used to populate status.oauthMetadata.
+	// The key "oauthMetadata" is used to locate the data.
+	// If specified and the config map or expected key is not found, no metadata is served.
+	// If the specified metadata is not valid, no metadata is served.
 	// +optional
-	OAuthMetadata ConfigMapReference `json:"oauthMetadata"`
+	OAuthMetadata ConfigMapNameReference `json:"oauthMetadata"`
 
 	// webhookTokenAuthenticators configures remote token reviewers.
 	// These remote authentication webhooks can be used to verify bearer tokens
 	// via the tokenreviews.authentication.k8s.io REST API.  This is required to
 	// honor bearer tokens that are provisioned by an external authentication service.
+	// If this list is non-empty, it is used to calculate the
+	// observed value stored in status.webhookTokenAuthenticators.
+	// Otherwise, observed cluster state may be used to populate status.webhookTokenAuthenticators.
+	// +optional
 	WebhookTokenAuthenticators []WebhookTokenAuthenticator `json:"webhookTokenAuthenticators"`
 }
 
 type AuthenticationStatus struct {
+	// type identifies the current user facing authentication mode in use.
+	// See authentication spec.type.
+	Type AuthenticationType `json:"type"`
```

Comment:
This usage is questionable.  Please remove.  It is novel and will be inconsistent with feature based configuration that drives multiple binaries.

Reply:
Does not look novel:

https://github.com/openshift/api/blob/84404551741398f2300f84b16fb97ef351749982/config/v1/types_network.go#L34-L52